lib/iris/tests/idiff.py

```diff
@@ -160,28 +166,45 @@ def step_over_diffs(result_dir):
             try:
                 im = Image.open(result_fname)
                 if im.format != 'PNG':
+                    # Ignore - it's not a png image.
                     continue
             except IOError:
+                # Ignore - it's not an image.
                 continue
             key = os.path.splitext('-'.join(result_fname.split('-')[1:]))[0]
             try:
-                uri = repo[key][0]
+                uri = repo[key][index]
             except KeyError:
                 wmsg = 'Ignoring unregistered test result {!r}.'
                 warnings.warn(wmsg.format(key))
                 continue
             with temp_png(key) as expected_fname:
                 processed = True
                 resource = requests.get(uri)
```

Comment:
`requests.head` is a bit faster than `requests.get` as it doesn't download the image file